docs/conf.py

```diff
@@ -195,25 +186,186 @@
+versions.append(("0.1.2", "dali_012_beta", "short_devel"))
+versions.append(("0.1.1", "dali_011_beta", "short_devel"))
+versions.append(("0.1", "dali_01_beta", "devel"))
+
+versions = sorted(versions, key=lambda v: Version(v[0]), reverse=True)
+
+json_data = []
+for v in versions:
+    if len(v) > 2 and v[2] == "devel":
+        json_data.append(
+            {
+                "name": v[0],
+                "version": v[0],
+                "url": f"https://docs.nvidia.com/deeplearning/dali/archives/{v[1]}"
+                "/dali-developer-guide/docs/",
+            }
+        )
+    elif len(v) > 2 and v[2] == "short_devel":
+        json_data.append(
+            {
+                "name": v[0],
+                "version": v[0],
+                "url": f"https://docs.nvidia.com/deeplearning/dali/archives/{v[1]}"
+                "/dali-developer-guide/",
+            }
+        )
+    elif len(v) > 2 and v[2] == "short_user":
+        json_data.append(
+            {
+                "name": v[0],
+                "version": v[0],
+                "url": f"https://docs.nvidia.com/deeplearning/dali/archives/{v[1]}/user-guide/",
+            }
+        )
+    else:
+        json_data.append(
+            {
+                "name": v[0],
+                "version": v[0],
+                "url": f"https://docs.nvidia.com/deeplearning/dali/archives/{v[1]}"
+                "/user-guide/docs/",
+            }
+        )
+
+if "dev" in version_long:
+    version_short_split = version_short.split(".")
+    one_before = f"{version_short_split[0]}.{int(version_short_split[1]) - 1}"
+    json_data.insert(
+        0,
+        {
+            "name": f"{one_before} (current release)",
+            "version": f"{one_before} (current release)",
+            "url": "https://docs.nvidia.com/deeplearning/dali/user-guide/docs/",
+        },
+    )
+else:
+    json_data.insert(
+        0,
+        {
+            "name": f"{version_short} (current release)",
+            "version": version_short,
+            "url": "https://docs.nvidia.com/deeplearning/dali/user-guide/docs/",
+        },
+    )
+
+json_data.insert(
+    1,
+    {
+        "name": "main (unstable)",
+        "version": "main",
+        "url": "https://docs.nvidia.com/deeplearning/dali/main-user-guide/docs/",
+    },
+)
+
+# trim to N last releases and add the archive
+json_data = json_data[0:10]
+
+json_data.append(
+    {
+        "name": "older releases",
+        "version": "archives",
+        "url": "https://docs.nvidia.com/deeplearning/dali/archives/index.html",
+    }
+)
+
+# validate links
+
+for i, d in enumerate(json_data):
+    if i == 2:
+        # as we just generate the switcher.json for the next release the one before is
+        # not in the archive yet skip checking it
+        print(f"skip checking not archived release location for the switcher: {d['url']}")
+        continue
+    h = httplib2.Http()
+    resp = h.request(d["url"], "HEAD")
```

Comment:
We probably need to append the index.html here at the end too, to check the exact url as presented by the switcher.

Reply:
Done.